source/non_matching/mesh_classifier.cc

```diff
@@ -0,0 +1,531 @@
+
+        /**
+         * Return the FECollection of the DoFHandler passed to the constructor.
+         */
+        const hp::FECollection<dim> &
+        get_fe_collection() const override;
+
+        /**
+         * Return the active_fe_index of the DoFCellAccessor associated with the
+         * DoFHandler and the the incoming cell in the triangulation.
+         */
+        unsigned int
+        active_fe_index(const typename Triangulation<dim>::active_cell_iterator
+                          &cell) const override;
+
+        /**
+         * Writes the local cell dofs of the global level set vector to
+         * @p local_levelset_values.
+         */
+        void
+        get_local_level_set_values(
+          const typename Triangulation<dim>::active_cell_iterator &cell,
+          Vector<double> &local_levelset_values) override;
+
+        /**
+         * Writes the local face dofs of the global level set vector to
+         * @p local_levelset_values.
+         */
+        void
+        get_local_level_set_values(
+          const typename Triangulation<dim>::active_cell_iterator &cell,
+          const unsigned int                                       face_index,
+          Vector<double> &local_levelset_values) override;
+
+      private:
+        /**
+         * Pointer to the DoFHandler associated with the level set function.
+         */
+        const SmartPointer<const DoFHandler<dim>> dof_handler;
+
+        /**
+         * Pointer to the vector global dof values of the level set function.
+         */
+        const SmartPointer<const VECTOR> level_set;
+      };
+
+
+
+      template <int dim, class VECTOR>
+      DiscreteLevelSetDescription<dim, VECTOR>::DiscreteLevelSetDescription(
+        const DoFHandler<dim> &dof_handler,
+        const VECTOR &         level_set)
+        : dof_handler(&dof_handler)
+        , level_set(&level_set)
+      {}
+
+
+
+      template <int dim, class VECTOR>
+      void
+      DiscreteLevelSetDescription<dim, VECTOR>::get_local_level_set_values(
+        const typename Triangulation<dim>::active_cell_iterator &cell,
+        Vector<double> &local_levelset_values)
+      {
+        typename DoFHandler<dim>::active_cell_iterator cell_with_dofs(
+          &dof_handler->get_triangulation(),
+          cell->level(),
+          cell->index(),
+          dof_handler);
+
+        // Get the dofs indices associated with the face.
+        const unsigned int n_dofs_per_cell =
+          cell_with_dofs->get_fe().n_dofs_per_cell();
+        std::vector<types::global_dof_index> dof_indices(n_dofs_per_cell);
+        cell_with_dofs->get_dof_indices(dof_indices);
+
+        local_levelset_values.reinit(dof_indices.size());
+
+        for (unsigned int i = 0; i < dof_indices.size(); i++)
+          local_levelset_values[i] =
+            dealii::internal::ElementAccess<VECTOR>::get(*level_set,
+                                                         dof_indices[i]);
```

Comment:
Shouldn't you be able to use `cell->get_dof_values(*level_set, local_levelset_values)`, see here: https://github.com/dealii/dealii/blob/4fc03d1aab8ed6c76da0ee6de67ac8a18c889b01/include/deal.II/dofs/dof_accessor.templates.h#L2656-L2661

Reply:
I had forgotten to remove this function. It is not needed now that we determine the `LocationToLevelSet` of the cell from the location of it's faces.